editor/src/components/canvas/dom-lookup.ts

```diff
@@ -149,6 +151,41 @@ export function getAllTargetsAtPoint(
     return []
   }
   const elementsUnderPoint = document.elementsFromPoint(point.x, point.y)
+  // TODO FIXME we should take the zero-sized elements from Canvas.getAllTargetsAtPoint, and insert them (in a correct-enough order) here. See PR for context https://github.com/concrete-utopia/utopia/pull/2345
+  return findFirstValidParentsForAllElements(validElementPathsForLookup, elementsUnderPoint)
+}
+
+const findFirstValidParentForSingleElement = memoize(findFirstValidParentForSingleElementUncached, {
```

Comment:
what equality function does memoize use by default? Asking because I think the `Array<Elements>` is something we haven't tried memoizing before. Maybe it'd make sense to provide our own equality function here? just so it's very visible what happens under the hood?

Reply:
We use our own `shallowEquals` by default: https://github.com/concrete-utopia/utopia/blob/master/editor/src/core/shared/memoize.ts#L10

And that uses our own `abstractEquals` which compares arrays by their values:
https://github.com/concrete-utopia/utopia/blob/master/editor/src/core/shared/equality-utils.ts#L11

https://github.com/concrete-utopia/utopia/blob/master/editor/src/core/shared/equality-utils.ts#L37

I can pass in `shallowEqual` by myself, but I think this has been deliberately provided as a good general default equality function for memoization, I don't really feel the need here to be honest.